contribute.md

```diff
@@ -0,0 +1,17 @@
+# Contribute
+
+go-ipfs is MIT licensed open source software. We welcome contributions big and
+small! Take a look at the [community contributing notes](https://github.com/ipfs/community/blob/master/contributing.md). Please make sure to check the [issues](https://github.com/ipfs/go-ipfs/issues). Search the closed ones
+before reporting things, and help us with the open ones.
+
+Guidelines:
+
+- see the [dev pseudo-roadmap](dev.md)
+- please adhere to the protocol described in [the main ipfs repo](https://github.com/ipfs/ipfs), [paper](http://static.benet.ai/t/ipfs.pdf), and [specs](https://github.com/ipfs/specs) (WIP).
+- please make branches + pull-request, even if working on the main repository
+- ask questions or talk about things in [Issues](https://github.com/ipfs/go-ipfs/issues) or #ipfs on freenode.
+- ensure you are able to contribute (no legal issues please-- we'll probably setup a CLA)
+- run `go fmt` before pushing any code
+- run `golint` and `go vet` too -- some things (like protobuf files) are expected to fail.
```

Comment:
>  are there more than this one go repo?

yeah, we have many go repos. i think we should have `<lang>-contribute.md` and put all lang specific things there, and yep, link to that from this file (so it gets a notcie on posting issues)